### PR TITLE
new billion scale blog post

### PR DESCRIPTION
**Pull request by jobergum**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: _posts/2022-10-07-building-billion-scale-vector-search.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+models are on the rise, combining multiple modalities, for example, text and image data. These advances in deep learning
+leading to better vector representations of data have driven increased interest in searching these representations at
+scale. 
+
+![vectorization](/assets/2022-10-07-building-billion-scale-vector-search/ml-models.png)
+
+
+Any machine learning algorithm must transform the input and output data into a data format the machine understands. This
+is where vectors and, generally, tensors come into the picture. Everything can be represented as
+a vector in high dimensional vector space.
```
Comment (kkraune) [inline _posts/2022-10-07-building-billion-scale-vector-search.md:41]: Avoid repetition
```suggestion
is where vectors and, generally, tensors come into the picture.
```

File: _posts/2022-10-07-building-billion-scale-vector-search.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+store.
+
+
+## Building cost-efficient large-scale vector search 
+
+Over the past few years, we have made a few observations around scaling vector search to billions of data items:
+
+* First, surprisingly, many organizations have a lot of raw unstructured data, petabytes of data that easily reach
+* billions of rows. Second, AI models to generate vector representations from this data have become a commodity, thanks
+* to [Huggingface](https://huggingface.co/).
```
Comment (kkraune) [inline _posts/2022-10-07-building-billion-scale-vector-search.md:84]: There is something odd with this list - is this what you meant?
```suggestion
* Surprisingly, many organizations have a lot of raw unstructured data,
  petabytes of data that easily reach billions of rows.
* AI models to generate vector representations from this data have become a commodity,
  thanks to [Huggingface](https://huggingface.co/). 
```

**Review by kkraune — APPROVED** [review]
This is great, a super teaser into what is to come, and also a good high-level intro into the problem domain! Looking forward to the next posts!

File: _posts/2022-10-07-building-billion-scale-vector-search.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+### Efficient Vector Search
+
+Vespa implements a[ real-time version of the HNSW algorithm](https://docs.vespa.ai/en/approximate-nn-hnsw.html) for
+efficient and high-recall ANNS. The implementation is verified with multiple vector datasets on[
+ann-benchmarks.com](http://ann-benchmarks.com/) and used in production by
+[Spotify](https://engineering.atspotify.com/2022/03/introducing-natural-language-search-for-podcast-episodes/). 
+
+
+### Highly Extensible Architecture
+![architecture](/assets/2022-10-07-building-billion-scale-vector-search/architecture.png)
```
[the remark this answers is not in the document]
Comment (jobergum) [reply _posts/2022-10-07-building-billion-scale-vector-search.md]: Thanks, fixed

**Pull request merged**